src/sentry/replays/usecases/ingest/dom_index.py

```diff
@@ -300,27 +310,16 @@ def _should_report_hydration_error_issue(project_id: int) -> bool:
     ) and project.get_option("sentry:replay_hydration_error_issues")
 
 
-def _should_report_rage_click_issue(project_id: int) -> bool:
-    project = Project.objects.get(id=project_id)
-
-    def _project_has_feature_enabled() -> bool:
-        """
-        Check if the project has the feature flag enabled,
-        This is controlled by Sentry admins for release of the feature
-        """
-        return features.has(
-            "organizations:session-replay-rage-click-issue-creation",
-            project.organization,
-        )
-
-    def _project_has_option_enabled() -> bool:
-        """
-        Check if the project has the option enabled,
-        This is controlled by the project owner, and is a permanent setting
-        """
-        return project.get_option("sentry:replay_rage_click_issues")
-
-    return all([_project_has_feature_enabled(), _project_has_option_enabled()])
+def _should_report_rage_click_issue(project: Project) -> bool:
+    """
+    The feature is controlled by Sentry admins for release of the feature,
+    while the project option is controlled by the project owner, and is a
+    permanent setting
+    """
+    return features.has(
+        "organizations:session-replay-rage-click-issue-creation",
+        project.organization,
```

Comment:
Organization is still not cached.

Reply:
It looks like Django caches foreign key queries by default: https://docs.djangoproject.com/en/dev/topics/db/queries/#one-to-many-relationships ("Forward access to one-to-many..").

To do it all in one query (like a JOIN), I added `select_related`